test/modules/aws-ec2-gpv-integration.ts

```diff
@@ -119,6 +139,216 @@ describe('EC2 General Purpose Volume Integration Testing', () => {
+        (SELECT id FROM instance WHERE tags ->>'name' = '${prefix}-1' LIMIT 1));
+        
+      COMMIT;
+    `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    )((e?: any) => (!!e ? done(e) : done(undefined)));
+  });
+  it('creates the instance with specific volume', commit());
+
+  itDocs(
+    'check number of instances',
+    query(
+      `
+    SELECT *
+    FROM instance
+    WHERE tags ->> 'name' = '${prefix}-1';
+  `,
+      (res: any[]) => expect(res.length).toBe(1),
+    ),
+  );
+  itDocs(
+    'check number of volumes',
+    query(
+      `
+    SELECT *
+    FROM general_purpose_volume INNER JOIN instance_block_device_mapping ON general_purpose_volume.id = instance_block_device_mapping.volume_id
+    WHERE instance_block_device_mapping.instance_id = (SELECT id FROM instance WHERE tags ->> 'name' = '${prefix}-1');
+  `,
+      (res: any[]) => {
+        expect(res.length).toBe(1);
+        expect(res[0].size).toBe(15);
+      },
+    ),
+  );
+
+  it('starts a transaction', begin());
+  itDocs('creates and attaches an extra volume to instance', (done: (arg0: any) => any) => {
+    query(
+      `
+      BEGIN;
+        INSERT INTO general_purpose_volume (volume_type, size, availability_zone, tags)
+        VALUES ('gp2', 20, '${availabilityZone2}', '{"Name": "${gp2VolumeName}-2"}');
+        INSERT INTO instance_block_device_mapping (device_name, volume_id, instance_id, delete_on_termination) values ('/dev/sda2', (SELECT id FROM general_purpose_volume WHERE tags ->>'Name' = '${gp2VolumeName}-2' LIMIT 1),
+        (SELECT id FROM instance WHERE tags ->>'name' = '${prefix}-1' LIMIT 1), FALSE);
+        
+      COMMIT;
+    `,
+      undefined,
+      true,
+      () => ({ username, password }),
+    )((e?: any) => (!!e ? done(e) : done(undefined)));
+  });
+  it('attaches extra volume', commit());
+
+  itDocs(
+    'check extra volume has been created',
+    query(
+      `
+    SELECT *
+    FROM general_purpose_volume INNER JOIN instance_block_device_mapping ON general_purpose_volume.id = instance_block_device_mapping.volume_id
+    WHERE instance_block_device_mapping.instance_id = (SELECT id FROM instance WHERE tags ->> 'name' = '${prefix}-1');
+  `,
+      (res: any[]) => {
+        expect(res.length).toBe(2);
+      },
+    ),
+  );
+
+  it('starts a transaction', begin());
+
+  itDocs('creates an instance with specific size and an extra volume', (done: (arg0: any) => any) => {
+    query(
+      `
+      BEGIN;
+        INSERT INTO general_purpose_volume (volume_type, size, availability_zone, tags, is_root_device)
+        VALUES ('gp2', 15, '${availabilityZone2}', '{"Name": "${gp2VolumeName}-test-1"}', TRUE);
+        INSERT INTO general_purpose_volume (volume_type, size, availability_zone, tags, is_root_device)
+        VALUES ('gp2', 20, '${availabilityZone2}', '{"Name": "${gp2VolumeName}-test-2"}', TRUE);
```

Comment:
This can be

```suggestion
        INSERT INTO general_purpose_volume (volume_type, size, availability_zone, tags, is_root_device)
        VALUES ('gp2', 15, '${availabilityZone2}', '{"Name": "${gp2VolumeName}-test-1"}', TRUE),
        VALUES ('gp2', 20, '${availabilityZone2}', '{"Name": "${gp2VolumeName}-test-2"}', TRUE);
```